kubeadm/v1.15.0/KubeCluster.ps1

```diff
@@ -0,0 +1,273 @@
+Param(
+    [parameter(Mandatory = $false,HelpMessage="Print the help")]
+    [switch] $help,
+    [parameter(Mandatory = $false,HelpMessage="Install pre-requisites")]
+    [switch] $install,
+    [parameter(Mandatory = $false,HelpMessage= "Join the windows node to the cluster")]
+    [switch] $join,
+    [parameter(Mandatory = $false,HelpMessage="Reset and clean up this Windows worker node")]
+    [switch] $reset,
+    [parameter(Mandatory = $false,HelpMessage="Path to input configuration JSON")] 
+    $ConfigFile
+)
+
+function Usage()
+{
+    $bin = $PSCommandPath 
+    Get-Help $bin -Detailed
+
+    $usage = "
+    Usage: 
+		$bin [-help] [-init] [-join] [-reset]
+
+	Examples:
+        $bin -help                                                           Prints this help
+        $bin -install -ConfigFile kubecluster.json                           Sets up this Windows node to run containers
+        $bin -join -ConfigFile kubecluster.json                              Joins the Windows node to existing cluster
+        $bin -reset -ConfigFile kubecluster.json                             Resets and cleans up this Windows worker node
+    "
+
+    Write-Host $usage
+}
+
+# Handle --help
+if ($help.IsPresent)
+{
+    Usage
+    exit
+}
+
+function ReadKubeclusterConfig($ConfigFile)
+{
+    # Read the configuration and initialize default values if not found
+    $Global:ClusterConfiguration = ConvertFrom-Json ((GetFileContent $ConfigFile -ErrorAction Stop) | out-string)
```

Comment:
Would it be better to use Try/Catch to tell the user what is wrong with the JSON? rather than relying on them to be able to decipher the PowerShell error?

Reply:
Let's add this in a future PR so this one doesn't get held up